feat: setup home page

vgeorge opened the pull request:
Contributes to https://github.com/NASA-IMPACT/veda-ui/issues/1517.

This is a work in progress, opening for visibility.

To display a responsive layout where the image occupies the full available height, we need to set an absolute height on the section’s parent. It is calculated on the client by querying the header and footer heights, and applied to the main element.

Next steps:

- Add styles to the title and button
- Enhance the loading state


cc @snmln

snmln commented:
@vgeorge I think this looks really good so far! I believe we might need to get the netlify preview setup within this repo. I looked into doing so on my own but it appears that the netlify functionality is hosted under a devseed account. Not sure if we can handle that ourselves or need to loop in Gjore?

vgeorge commented:
@snmln thanks for looking into this. It seems the approach I used is not working well on all browsers. It would be nice to find a solution that doesn't rely on `useEffect`, so I'll explore a bit more the options. I believe we can create deploy previews to Devseed's Vercel account, I'll add a ticket.

vgeorge commented:
Changes added:

- Removed client code to determine section height in favor of CSS approach
- Changed property `$theme-color-primary: 'red-50'` as it doesn't seem to be applied to this instance
- Changed button radius property to `$theme-button-border-radius: 0`
- Added animated transition from data themes
- Added 'Get started' button

Notes:

- It is suggested that the header and footer have a minimal height to avoid visible layout changes when l loading
- SVG icon from USWDS was added inline due to incompability of react-uswds with Next.js build
- I believe we should look into setting up styles-components to brind styles closer to components

Preview:

![preview](https://github.com/user-attachments/assets/d15ec035-8159-4a69-9c6c-53f251ef2574)


This is ready for review.

AliceR reviewed (approved):
The loading header seems so weird to me! Its such a tiny component, it should just render straight away... We should investigate it more!

The page looks great! Awesome that you found a css only approach for the layout.
Out of curiosity: Why did you decide to use a css file over utility classes? Since `minh-viewport` is a utility class from uswds, it seems unintuitive to me to overwrite that in homepage.scss. But I am also generally not a fan of separating styles from their components, so thats a bit philosophical :D 

I love your git hygiene! Just curious about the conventional commit - I thought we would only use the keywords on the PR title, due the release automation. Thats for veda-ui at least, I assumed the same for the instances for consistencies sake. 
Also, would you consider cleaning up the commit history with an interactive rebase? 

Just thoughts, overall this looks good!

dzole0311 commented on app/page.tsx:
```diff
@@ -1,105 +1,59 @@
-const otherStories = featuredStories.slice(1);
+const FADE_DURATION = 3000; // Duration in milliseconds
 
 export default function HomePage() {
```
I found an issue when moving through routes:

1. Start from the home page
2. Click on Data Catalog in the Header
3. The home page component is still shown

vgeorge replied:
This should be fixed now.

vgeorge commented:
@AliceR @dzole0311 thanks for your reviews!

I added some changes, some notes:

- Tiny header: this is due to the way the placeholder header was implemented. I scoped this PR to the contents of the home page, the header and footer should be addressed in separate (see https://github.com/NASA-IMPACT/veda-ui/issues/1518)
- Git commits: @AliceR I'm not sure if I understood your question, I think the commits are already following conventional commits. Do you mean renaming the PR title?
- `minh-viewport` was removed. I moved these styles to the layout, but lets improve this further once we have styles-components working

This is ready for another review.

dzole0311 reviewed (approved):
Lgtm!

dzole0311 commented:
@vgeorge Sorry for that, yes, maybe let's point it to a develop branch - I've just created a ruleset for it. Let me know if it needs tweaking.
